libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -1012,7 +1021,7 @@ proc handleGraft(g: GossipSub,
 
       continue
 
-    if peer notin g.peerStats:
+    if peer.peerId notin g.peerStats:
       g.onNewPeer(peer)
```

Comment:
this looks odd - `onNewPeer` becomes ambiguous as to whether it's a new peer subscribing or new to stats - would be better to move this to a `initPeerStats()` function that gets called from both `onNewPeer` and here